### PR TITLE
Move the entry point to start_terminal_interface

### PR DESCRIPTION
**Pull request by Notnaton**
While talking with @Arrendy going through some code made a change to the entry point
This is building further on the comments made in https://github.com/KillianLucas/open-interpreter/pull/947

### Pre-Submission Checklist (optional but appreciated):

- [ ] I have included relevant documentation updates (stored in /docs)
- [x] I have read `docs/CONTRIBUTING.md`
- [ ] I have read `docs/ROADMAP.md`

### OS Tests (optional but appreciated):

- [x] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux

### CODE REVIEW

**Review by Arrendy — APPROVED** [review]
Looks great! Definitely more logical having this in `start_terminal_interface.py`

File: interpreter/terminal_interface/start_terminal_interface.py [hunk 1]
```diff
@@ -689,3 +690,11 @@ def start_terminal_interface(interpreter):
...
 
     interpreter.chat()
+
+
+def main():
+    interpreter = OpenInterpreter()
+    try:
+        start_terminal_interface(interpreter)
+    except KeyboardInterrupt as e:
+        print(e)
```
Comment (Arrendy) [inline interpreter/terminal_interface/start_terminal_interface.py:700]: I believe `e` will always be an empty string. Do we want to revert to `Exited.` or some other message to show the user?

Comment (Notnaton) [reply interpreter/terminal_interface/start_terminal_interface.py]: It does print something, no?
Anyways, I added more Error handling and a finally:

Comment (Arrendy) [reply interpreter/terminal_interface/start_terminal_interface.py]: It didn't print anything in my testing and chatGPT says it'll always be empty. I'm no python expert though. Did it print a message in your testing?

**Comment by KillianLucas** [comment]
Nice!! Thank you so much @Notnaton. Yes, this is much more logical. Fantastic work, merging now.

(Also: really looking forward to our chat tomorrow!)

**Pull request merged**